### PR TITLE
Avoid re-creating the encoders WriteCloser for every request

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: config/confighttp/compressor.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+		if closeErr != nil {
+			return closeErr
+		}
+	}
+
+	if err := writer.Close(); err != nil {
+		return err
+	}
+	p.pool.Put(writer)
```
Comment (dmitryax) [inline config/confighttp/compressor.go:76]: Shouldn't we have `defer p.pool.Put(writer)` given that we have returns above?

Comment (bogdandrutu) [reply config/confighttp/compressor.go:76]: I thought about this, but I don't know if it is safe to return a failed writer... If we don't return GC will collect this as any other object.

Comment (bogdandrutu) [reply config/confighttp/compressor.go]: Tried to read some of the implementations gzip and snappy and seem to be fine...

**Review by dmitryax — APPROVED** [review]
LGTM

**Pull request merged**